File: Java solutions/Arrays/containsDuplicate.java
import java.util.*;
class Solution {
    public boolean containsDuplicate(int[] nums) {

        // Initialize a set to keep record of all the numbers encountered till now.
        
        Set<Integer> set = new HashSet<>();

        // Iterate over all the numbers in the given array and try adding them to the set above.
        // If add fails(element already exits in the set) it'll return false, which indicates
        // duplicate and hence we return true.

        // If add is a success (element is not present in the set) it'll return true, we continue.

        for(int i : nums)
        {
            if(!set.add(i))
            {
                return true;
            }
        }

        //If we are able to go through all elements without duplicates, we return false.
        return false;
        
    }
}

File: Python solutions/Arrays/contains_duplicate.py
from ast import List
class Solution:
    def containsDuplicate(self, nums: List[int]) -> bool:
        
        # Initialize a dict
        arr = {}

        # Iterate over all nums in the list and check if num is already in dict.
        # If yes return True else add it to dict with its count as 1.
        # If you iterate over all elements without returning true return false.
        
        for i in nums:
            if i in arr:
                return True
            arr[i] = 1  
        
        return False
